Observer/CreditmemoObserver.php

```diff
@@ -0,0 +1,118 @@
+ *
+ * Copyright (c) 2022 Adyen N.V. (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Observer;
+
+use Adyen\Payment\Helper\AdyenOrderPayment;
+use Adyen\Payment\Helper\Creditmemo as CreditMemoHelper;
+use Adyen\Payment\Api\Data\OrderPaymentInterface;
+use Adyen\Payment\Helper\Config;
+use Adyen\Payment\Helper\Invoice as InvoiceHelper;
+use Adyen\Payment\Helper\Order as OrderHelper;
+use Adyen\Payment\Logger\AdyenLogger;
+use Adyen\Payment\Model\Order\PaymentFactory;
+use Adyen\Payment\Model\ResourceModel\Order\Payment;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Exception\AlreadyExistsException;
+use Magento\Sales\Model\Order;
+use Magento\Sales\Model\Order\Invoice;
+use Magento\Sales\Model\Order\StatusResolver;
+use Adyen\Payment\Helper\PaymentMethods;
+
+class CreditmemoObserver implements ObserverInterface
+{
+    /** @var Payment $adyenPaymentResourceModel */
+    private $adyenPaymentResourceModel;
+
+    /** @var PaymentFactory */
+    private $adyenOrderPaymentFactory;
+
+    /** @var InvoiceHelper $invoiceHelper*/
+    private $invoiceHelper;
+
+    /** @var StatusResolver $statusResolver */
+    private $statusResolver;
+
+    /** @var AdyenOrderPayment $adyenOrderPaymentHelper */
+    private $adyenOrderPaymentHelper;
+
+    /** @var CreditmemoHelper $creditmemoHelper */
+    private $creditmemoHelper;
+
+    /** @var Config $configHelper */
+    private $configHelper;
+
+    /** @var PaymentMethods $paymentMethodsHelper */
+    private $paymentMethodsHelper;
+
+    /** @var OrderHelper */
+    private $orderHelper;
+
+    /**
+     * @var AdyenLogger
+     */
+    private $logger;
+
+    public function __construct(
+        Payment           $adyenPaymentResourceModel,
+        PaymentFactory    $adyenOrderPaymentFactory,
+        InvoiceHelper     $invoiceHelper,
+        StatusResolver    $statusResolver,
+        AdyenOrderPayment $adyenOrderPaymentHelper,
+        CreditmemoHelper $creditmemoHelper,
+        Config            $configHelper,
+        AdyenLogger       $adyenLogger,
+        PaymentMethods    $paymentMethodsHelper,
+        OrderHelper       $orderHelper
+    ) {
+        $this->adyenPaymentResourceModel = $adyenPaymentResourceModel;
+        $this->adyenOrderPaymentFactory = $adyenOrderPaymentFactory;
+        $this->invoiceHelper = $invoiceHelper;
+        $this->statusResolver = $statusResolver;
+        $this->adyenOrderPaymentHelper = $adyenOrderPaymentHelper;
+        $this->creditmemoHelper = $creditmemoHelper;
+        $this->configHelper = $configHelper;
+        $this->logger = $adyenLogger;
+        $this->paymentMethodsHelper = $paymentMethodsHelper;
+        $this->orderHelper = $orderHelper;
+    }
+
+    /**
+     * Link all adyen_invoices to the appropriate magento invoice and set the order to PROCESSING to allow
```

Comment:
Can we update this comment (I think it should reference adyen_creditmemos and not adyen_invoices)